README.md

```diff
@@ -46,27 +46,50 @@ HttpProxyServer server =
                   }
 
                   @Override
-                  public HttpResponse proxyToServerRequest(HttpObject httpObject) {
-                      // TODO: implement your filtering here
-                      return null;
-                  }
-
-                  @Override
                   public HttpObject serverToProxyResponse(HttpObject httpObject) {
                       // TODO: implement your filtering here
                       return httpObject;
                   }
-
-                  @Override
-                  public HttpObject proxyToClientResponse(HttpObject httpObject) {
-                      // TODO: implement your filtering here
-                      return httpObject;
-                  }   
-               };
             }
         })
         .start();
-```                
+```
+
+Please have a look into this base base class to see the hooks you can use. 
+
+To enable aggregator and inflater you have to return a value greater than 0 in 
+your `HttpFiltersSource#get(Request/Response)BufferSizeInBytes()` methods. This 
+provides to you a `FullHttp(Request/Response)' with the complete content in your 
+filter uncompressed. Otherwise you have to handle the chunks yourself.
+
+```java
+    @Override
+    public int getMaximumResponseBufferSizeInBytes() {
+        return 10 * 1024 * 1024;
+    }
+```
+
+This size limit applies to every connection. To disable aggregating by URL at 
+*.iso or *dmg files for example, you can return in your filters source a filter 
+like this:
+
+```java
+return new HttpFiltersAdapter(originalRequest, serverCtx) {
+    @Override
+    public void proxyToServerConnectionSucceeded(ChannelHandlerContext serverCtx) {
+        ChannelPipeline pipeline = serverCtx.pipeline();
```

Comment:
This is a very interesting solution to enable LP users to configure the Netty channel handler context directly. At some point, when we think about refactoring the connection flow and enhancing the filters further, I wonder if there's a way to make this more configurable through LP, rather than forcing users to interact with Netty directly? No obvious solution comes to mind, but I can definitely see the utility in the very large file use case you pointed out of being able to turn on/off aggregation on a per-request basis.

Anyway, very clever solution!